### PR TITLE
CLN: Address MulitIndex Test Follow Ups in Issue #21918

### LINKED ISSUE #21918: CLN: Addtional Refactoring of MultiIndex Tests
#### This is a follow up to  #18644 #21514

The referenced issue and pull request split the MultiIndex tests into separate files and introduced additional fixtures.  There is still outstanding comments from @jreback  to finalize the cleanup.

### PR DESCRIPTION
**Pull request by elmq0022**
- [x] closes #21918
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry - N/A

Addresses most of the issues.  Left blowing away the old pickles for another issue which is still open.

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
also merged #21900 so will need to rebase

**Comment by elmq0022** [comment]
@jreback all green.

File: pandas/tests/indexes/multi/test_analytics.py [hunk 1]
```diff
@@ -1,8 +1,437 @@
...
+                      pd.MultiIndex.from_tuples(idx))
+
+    tm.assert_series_equal(left, right)
+
+
+def test_bounds(idx):
+    idx._bounds
+
+
+def test_append(idx):
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_analytics.py]: can you move things like append / insert (ops that are shape shifting) to a new file, maybe test_reshape.py

File: pandas/tests/indexes/multi/test_analytics.py [hunk 2]
```diff
@@ -1,8 +1,437 @@
...
+        pytest.raises(NotImplementedError, f)
+
+
+def test_reorder_levels(idx):
+    # this blows up
+    tm.assert_raises_regex(IndexError, '^Too many levels',
+                           idx.reorder_levels, [2, 1, 0])
+
+
+@pytest.mark.parametrize('ordered', [True, False])
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_analytics.py]: we have an astype file (you just created)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/multi/test_analytics.py (hunk 1 above)
Comment (elmq0022) [reply pandas/tests/indexes/multi/test_analytics.py]: @jreback  would you put groupby into this category?

Comment (jreback) [reply pandas/tests/indexes/multi/test_analytics.py]: you can leave groupby here

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by elmq0022** [comment]
@jreback all green.  I made the additional edits and some further cleaning as well.  Thanks!

File: pandas/tests/indexes/multi/test_set_ops.py [hunk 3]
```diff
@@ -85,63 +64,37 @@ def test_difference_base(idx):
...
-    else:
-        answer = idx[[0, -1]]
-        result = first.symmetric_difference(second)
-        assert tm.equalContents(result, answer)
+    answer = idx[[0, -1]]
+    result = first.symmetric_difference(second)
+    assert tm.equalContents(result, answer)
 
     # GH 10149
     cases = [klass(second.values)
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_set_ops.py]: if you want to come back and parameterize things like this would be great (but future PR)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @elmq0022 nice cleaning!

**Pull request merged**